File: data/epl.py
from requests import get
from bs4 import BeautifulSoup
import json


class scrape_table:

    def __init__(self,index):
        request = get('https://en.wikipedia.org/wiki/2018%E2%80%9319_Premier_League')
        soup = BeautifulSoup(request.text,'html.parser')

        self.table = soup.find_all('table', class_='wikitable')[index]
        self.players = self.table.find_all('tr')[1:]

    
    def remove(self, s):
        s = s.strip()
        s = s.strip('\n')
        return s


    def is_int(self, string):
        try:
            int(string)
            return True
        except:
            return False


    def get_list(self):
        data = []

        for player in self.players:
            stats = player.find_all('td')
            stats = [self.remove(i.text) for i in stats]
            try:
                img = player.find_all('img')[0]
            except IndexError:
                break
            stats.append(img['alt'])
            stats.append('https:' + img['src'])
            data.append(stats)
            
            
        for i in data:
            if self.is_int(i[0]):
                del i[0]
            if self.is_int(i[2]):
                goals = i[2]
            else:
                i.insert(2,int(goals))
                
        return data


    def get_obj(self):
        x = self.get_list()
        obj = []
        for i in x: 
            obj.append({
                'name': i[0],
                'club': i[1],
                'amount': i[2],
                'nation': i[3],
                'flag': i[4]
            })
        return obj

goals = scrape_table(5)
assists = scrape_table(7)

with open('epl_goals.json', 'w') as f:
    json.dump(goals.get_obj(), f, indent=2)

with open('epl_assists.json', 'w') as f:
    json.dump(assists.get_obj(), f, indent=2)

File: script/epl.js
function createTable(tableID, jsonURL) {
  const table = document.getElementById(tableID);

  let count = 1;
  $.getJSON(jsonURL, function(stats) {
    
  for (var i = 0; i < stats.length; i ++) {
      let player = stats[i]
      
      row = table.insertRow(count);
      count ++;

      cell1 = row.insertCell(0)

      let flag = document.createElement('img')
      flag.src = player.flag;
      flag.alt = player.nation;
      flag.title = player.nation;
      $(cell1).append(flag);

      let name = document.createTextNode(player['name'])
      let space = document.createTextNode(' ');
      $(cell1).append(space);
      $(cell1).append(name);

      let cell2 = row.insertCell(1);
      let club = document.createTextNode(player.club);
      $(cell2).append(club);

      let cell3 = row.insertCell(2);
      let amount = document.createTextNode(player.amount);
      $(cell3).append(amount);
    }
  });
}

createTable('goalsTable', 'https://footballstats.github.io/data/epl_goals.json');
createTable('assistsTable', 'https://footballstats.github.io/data/epl_assists.json');